### PR TITLE
Fixes #908: Add support for multi-line environment files

### LINKED ISSUE #908: Multi-line environment files are not correctly processed
**Describe the bug**
For environment files, podman-compose passes them to `podman` with its `--env-file` option. However, `podman/docker` and `docker-compose` process environment files differently: `podman/docker` doesn't support multi-line environment files, while docker compose supports them (https://github.com/containers/podman/issues/18724#issuecomment-1757162444). 

**Expected behavior**
Properly support multi-line env files

**Actual behavior**
Just passes env-files to podman, which doesn't support multi-line env vars, as apparently docker also doesn't support them too.

** Implementation **
I found that it can be easily implemented (parse env file with dotenv, and pass variables to podman), and I'll open a PR to fix it too.

### PR DESCRIPTION
**Pull request by hedayat**
TBH, I don't find the part of compose spec which specifies this. Actually, it might not be according to the spec. However, this is what docker compose supports... !

### CODE REVIEW

File: podman_compose.py [hunk 1]
```diff
@@ -987,7 +987,12 @@ async def container_to_args(compose, cnt, detached=True):
         env_file = [env_file]
     for i in env_file:
         i = os.path.realpath(os.path.join(dirname, i))
-        podman_args.extend(["--env-file", i])
+        dotenv_dict = {}
```
Comment (p12tic) [inline podman_compose.py:990]: Needs a test. See pytests directory

**Review by p12tic — CHANGES_REQUESTED** [review]
Looks good, but needs tests

**Comment by p12tic** [comment]
This PR has been merged via https://github.com/containers/podman-compose/pull/949.

**Comment by mokibit** [comment]
Tests added in https://github.com/containers/podman-compose/pull/949.

**Pull request closed without merging**